python_modules/dagster/dagster/_grpc/impl.py

```diff
@@ -441,26 +455,27 @@ def get_partition_tags(
     instance_ref: Optional[InstanceRef] = None,
 ):
     try:
-        partition_set_def = repo_def.get_partition_set_def(partition_set_name)
+        (
+            job_def,
+            partitions_def,
+            partitioned_config,
+        ) = _get_job_partitions_and_config_for_partition_set_name(repo_def, partition_set_name)
 
         # Certain gRPC servers do not have access to the instance, so we only attempt to instantiate
         # the instance when necessary for dynamic partitions: https://github.com/dagster-io/dagster/issues/12440
-        if _partitions_def_contains_dynamic_partitions_def(partition_set_def.partitions_def):
-            with DagsterInstance.from_ref(
-                instance_ref
-            ) if instance_ref else nullcontext() as instance:
-                partition = partition_set_def.get_partition(
-                    partition_name, dynamic_partitions_store=instance
-                )
-        else:
-            partition = partition_set_def.get_partition(
-                partition_name, dynamic_partitions_store=None
-            )
+        instance = (
+            DagsterInstance.from_ref(instance_ref)
+            if (_partitions_def_contains_dynamic_partitions_def(partitions_def) and instance_ref)
+            else None
+        )
```

Comment:
DagsterInstance is a contextmanager that does useful things on teardown for performance - this should leave the `with` statement from before (or use an ExitStack) rather than relying on garbage collection to eventually clean it up

see also https://github.com/dagster-io/dagster/pull/13310/files

Reply:
Good to know, will post a PR to fix